tests/testthat/test-stat_labels.R

```diff
@@ -0,0 +1,124 @@
+test_that("ard_continuous() labels", {
+
+  # formula
+  ard_test <- ard_continuous(data = ADSL,
+                             by = "ARM",
+                             variables = c("AGE","BMIBL"),
+                             stat_labels = everything() ~ list(c("min","max") ~ "min - max"))
+
+  ard_test_labs <- ard_test %>%
+    dplyr::filter(stat_name %in% c("min","max")) |>
+    dplyr::pull(stat_label) |>
+    unique()
+
+  expect_equal(ard_test_labs,
+               "min - max")
+
+  # list
+  ard_test <- ard_continuous(data = ADSL,
+                             by = "ARM",
+                             variables = c("AGE","BMIBL"),
+                             stat_labels = everything() ~ list(p25 = "25th %ile", p75 = "75th %ile"))
+
+  actual_labs <- ard_test %>%
+    dplyr::filter(stat_name %in% c("p25","p75")) |>
+    dplyr::select(stat_name, stat_label) |>
+    dplyr::arrange(stat_name, stat_label) |>
+    unique()
+
+  expected_labs <- dplyr::tibble(
+    stat_name = c("p25","p75"),
+    stat_label = c("25th %ile","75th %ile")
+  )
+
+  expect_equal(actual_labs,
+              expected_labs,
+              ignore_attr = "class")
+
+  # variable-specific
+  ard_test <- ard_continuous(data = ADSL,
+                             by = "ARM",
+                             variables = c("AGE","BMIBL"),
+                             stat_labels = AGE ~ list(p25 = "25th %ile", p75 = "75th %ile"))
+
+  actual_labs <- ard_test %>%
+    dplyr::filter(stat_name %in% c("p25","p75")) |>
+    dplyr::select(variable, stat_name, stat_label) |>
+    dplyr::arrange(variable, stat_name, stat_label) |>
+    unique()
+
+  expected_labs <- dplyr::tribble(
+    ~ variable  ,  ~stat_name, ~stat_label,
+    "AGE"       , "p25"        , "25th %ile"  ,
+    "AGE"       , "p75"        , "75th %ile"     ,
+    "BMIBL"     , "p25"        , "25th Percentile",
+    "BMIBL"     , "p75"        , "75th Percentile"
+  )
+
+  expect_equal(actual_labs, expected_labs, ignore_attr = "class")
```

Comment:
Can you add one test like this where the summary function returns a named list of stats?

```r
expect_snapshot({
  conf_int <- function(x) t.test(x) |> broom::tidy() |> dplyr::select("conf.low", "conf.high") |> as.list()
  ard_continuous(
    ADSL, 
    variables = "AGE",
    statistics = ~list(conf.int = conf_int),
    stat_labels = ~list(conf.low = "LB", conf.high = "UB")
  ) |> 
    dplyr::select(variable, stat_name, stat_label)
})
```